legal-api/src/legal_api/models/db.py

```diff
@@ -11,17 +11,222 @@
+        try:
+            from legal_api.services import flags
+            db_versioning = flags.value('db-versioning')
+            enable_new_versioning = db_versioning.get('enable-new-versioning')
+            use_new_versioning = (bool(db_versioning.get('initialize-new-versioning'))
+                                  and bool(enable_new_versioning.get('legal-api')))
+
+            print(f"DB versioning flag values: {db_versioning}")
+            print(f"Use new versioning: {use_new_versioning}")
+            print(f"Current versioning: {session.current_versioning}")
+
+            # Update the session's versioning type based on the current flag value
+            session.enable_versioning('new' if use_new_versioning else 'old')
+
+            if session.is_new_versioning_active():
+                transaction_id = session.get_or_create_transaction()
+                for obj in session.new.union(session.dirty).union(session.deleted):
+                    if hasattr(obj, '__versioned_cls__'):
+                        print(f"Calling _before_flush for object: {obj.__class__.__name__} (id={getattr(obj, 'id', None)})")
+                        try:
+                            obj._before_flush(session, flush_context, instances)
+                        except Exception as e:
+                            print(f"Error in _before_flush for object {obj.__class__.__name__} (id={getattr(obj, 'id', None)}): {str(e)}")
+            elif session.is_old_versioning_active():
+                print("Using old versioning system (SQLAlchemy-Continuum)")
+                uow = versioning_manager.unit_of_work(session)
+                uow.process_before_flush(session)
+
+            print(f"New versioning active: {session.is_new_versioning_active()}")
+        except Exception as e:
+            print(f"Error in before_flush: {str(e)}")
+            import traceback
+            print(traceback.format_exc())
+        print("Exiting before_flush")
+
+    @event.listens_for(session, "after_flush")
+    def after_flush(session, flush_context):
+        print("Entering after_flush")
+        try:
+            if session.is_old_versioning_active():
+                print("Processing after_flush for old versioning")
+                uow = versioning_manager.unit_of_work(session)
+                uow.process_after_flush(session)
+        except Exception as e:
+            print(f"Error in after_flush: {str(e)}")
+            import traceback
+            print(traceback.format_exc())
+        print("Exiting after_flush")
+
+    @event.listens_for(session, "after_commit")
+    @event.listens_for(session, "after_rollback")
+    def clear_transaction(session):
+        print("Entering clear_transaction")
+        session.clear_transaction()
+
+    return session
+
+
+def initialize_versioning(app):
+    with app.app_context():
+        versioned_session(db.session)
+    print("Versioning initialized")
```

Comment:
db session initialized from here.  this leads to evaluation of db versioning flag and ensuring all the right things are enabled or disabled